fix: (Platform) table enhancements

olshansky opened the pull request:
#### Please provide a link to the associated issue.
Part of https://github.com/sap/fundamental-ngx/issues/4735
#### Please provide a brief summary of this pull request.
- Fixed row selection by Space/Enter button in single & multiple modes
- Fixed `ERROR Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked.`
- Added ability to scroll table in "page scroll" by keyboard 
- Minor a11y enhancements _(`headers` for cells, `aria-colindex/rowindex`, roles, scopes)_
- Fixed anchors on the examples page  
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [n/a] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

Betrozov commented on libs/platform/src/lib/components/table/directives/table-cell-selectable.directive.ts:
```diff
@@ -0,0 +1,18 @@
+import { AfterContentChecked, AfterViewChecked, Directive, HostListener } from '@angular/core';
+import { KeyUtil } from '@fundamental-ngx/core';
+import { DOWN_ARROW, PAGE_DOWN, PAGE_UP, SPACE, UP_ARROW } from '@angular/cdk/keycodes';
```
```suggestion
import { SPACE } from '@angular/cdk/keycodes';
```

sKudum commented:
Hi @olshansky,

Thank You for the fix. hope remaning wll be covered in future PRs,

Few points in current PR noticiable issues new.

1).On tab focus is not moving to row.
2). On dropdown select in dialog parent screen is displaying the dropdown values.
<img width="1336" alt="2021-02-24_22-14-30" src="https://user-images.githubusercontent.com/53487468/109034601-d2bf2100-76ed-11eb-8441-97dce9ef03ce.png">
3).All table are showing blue focus.
<img width="1453" alt="2021-02-24_22-17-49" src="https://user-images.githubusercontent.com/53487468/109034978-2fbad700-76ee-11eb-846a-149578d6a3dd.png">

fkolar commented:
> Hi @olshansky,
> 
> Thank You for the fix. hope remaning wll be covered in future PRs,
> 
> Few points in current PR noticiable issues new.
> 
> 1).On tab focus is not moving to row.
> 2). On dropdown select in dialog parent screen is displaying the dropdown values.
> <img alt="2021-02-24_22-14-30" width="1336" src="https://user-images.githubusercontent.com/53487468/109034601-d2bf2100-76ed-11eb-8441-97dce9ef03ce.png">
> 3).All table are showing blue focus.
> <img alt="2021-02-24_22-17-49" width="1453" src="https://user-images.githubusercontent.com/53487468/109034978-2fbad700-76ee-11eb-846a-149578d6a3dd.png">

@KevinOkamoto @dimamarksman, this is what we discussed today. What is the data representation of this filtering dialog so we can use it in our API.

olshansky commented:
> 1).On tab focus is not moving to row.
> 2). On dropdown select in dialog parent screen is displaying the dropdown values.
> 3).All table are showing blue focus.

@sKudum 
1)&nbsp; Do you mean each particular data row? If yes, let's move it to the "keyboard navigation" feature
1)&nbsp; fixed
1)&nbsp; fixed _(according to fiori design focus outline)_

dimamarksman commented on libs/platform/src/lib/components/table/table.component.ts:
```diff
@@ -576,6 +585,18 @@ export class TableComponent<T = any> extends Table implements AfterViewInit, OnD
+    keyDownHandler(event: KeyboardEvent): void {
+        if (this.loading) {
+            event.preventDefault();
+            event.stopPropagation();
```
do we really need stopPropagation?
